source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -0,0 +1,302 @@
+                                                       const ComponentMask &selected_properties,
+                                                       const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell) const
+      {
+        const unsigned int n_particle_properties = particle_handler.n_properties_per_particle();
+
+        const unsigned int property_index = selected_properties.first_selected_component(selected_properties.size());
+
+        AssertThrow(property_index != numbers::invalid_unsigned_int,
+                    ExcMessage("Internal error: the particle property interpolator was "
+                               "called without a specified component to interpolate."));
+
+        const Point<dim> approximated_cell_midpoint = std::accumulate (positions.begin(), positions.end(), Point<dim>())
+                                                      / static_cast<double> (positions.size());
+
+        typename parallel::distributed::Triangulation<dim>::active_cell_iterator found_cell;
+
+        if (cell == typename parallel::distributed::Triangulation<dim>::active_cell_iterator())
+          {
+            // We can not simply use one of the points as input for find_active_cell_around_point
+            // because for vertices of mesh cells we might end up getting ghost_cells as return value
+            // instead of the local active cell. So make sure we are well in the inside of a cell.
+            Assert(positions.size() > 0,
+                   ExcMessage("The particle property interpolator was not given any "
+                              "positions to evaluate the particle cell_properties at."));
+
+
+            found_cell =
+              (GridTools::find_active_cell_around_point<> (this->get_mapping(),
+                                                           this->get_triangulation(),
+                                                           approximated_cell_midpoint)).first;
+          }
+        else
+          found_cell = cell;
+
+        const typename ParticleHandler<dim>::particle_iterator_range particle_range =
+          particle_handler.particles_in_cell(found_cell);
+
+
+        std::vector<std::vector<double> > cell_properties(positions.size(),
+                                                          std::vector<double>(n_particle_properties,
+                                                                              numbers::signaling_nan<double>()));
+
+        const unsigned int n_particles = std::distance(particle_range.begin(), particle_range.end());
+
+        const unsigned int n_matrix_columns = (dim == 2) ? 6 : 10;
+        AssertThrow(n_particles >= n_matrix_columns,
+                    ExcMessage("At least one cell didn't contain enough particles to interpolate a unique solution for the cell."
+                               " The 'quadratic' interpolation scheme does not support this case."));
```

Comment:
That's something one could address in a later step: if there are not enough particles, one can always back down to a linear scheme or, in the worst case, a constant scheme. That would be in the spirit of a pseudoinverse to an operator.